### PR TITLE
Composite filter: Switch the order to create from factory context first

### PR DESCRIPTION
**Pull request by tyxia**
Switch the order to create the filter factory creation function from factory context first.

### CODE REVIEW

File: source/extensions/filters/http/composite/action.h [hunk 1]
```diff
@@ -43,29 +43,21 @@ class ExecuteFilterActionFactory
...
     }
 
+    // If above failed, try to create the filter factory creation function from server factory
+    // context (if exists).
+    if (callback == nullptr && context.server_factory_context_.has_value()) {
+      callback = factory.createFilterFactoryFromProtoWithServerContext(
+          *message, context.stat_prefix_, context.server_factory_context_.value());
+    }
+
+    RELEASE_ASSERT(callback != nullptr, "Failed to get filter factory creation function.");
```
Comment (yanavlasov) [inline source/extensions/filters/http/composite/action.h:59]: Can we throw here? Is this code running on the worker thread?

Comment (tyxia) [reply source/extensions/filters/http/composite/action.h]: done. I have to move this function to .cc file to throw exception as exception is not allowed in header filter in general.

PTAL, Thanks

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**